qubes/storage/reflink.py

```diff
@@ -321,9 +317,12 @@ async def import_volume(self, src_volume):
                 await self._import_data_end_unlocked(success)
         return self
 
-    def _path_revision(self, revision, timestamp=None):
+    def _path_revision(self, revision=None, timestamp=None):
         if timestamp is None:
-            timestamp = self.revisions[revision]
+            if revision is None:
+                revision, timestamp = list(self.revisions.items())[-1]
+            else:
+                timestamp = self.revisions[revision]
```

Comment:
This is not equivalent to the old code; consider the case where timestamp is not `None` but revision is.

Reply:
It's intentional, I've added a note to the commit message

Reply:
That makes sense.  I suggest including this in the function’s docstring, and throwing a better exception than `TypeError: can only concatenate str (not "NoneType") to str`.